src/views/LoginPage.vue

```diff
@@ -0,0 +1,147 @@
+<script setup lang="ts">
+import { computed, ref } from "vue";
+import { useUserStore } from "@/stores/user";
+import validateEmail from "@/helpers/validateEmail";
+import BaseInput from "@/components/base/BaseInput.vue";
+import BaseButton from "@/components/base/BaseButton.vue";
+import BaseCard from "../components/base/BaseCard.vue";
+import touchAll from "@/helpers/touchAll";
+import type { LoginCredentials } from "@/types/user";
+const userStore = useUserStore();
+
+const loginFormData = ref<LoginCredentials>({
+  email: "",
+  password: "",
+});
+
+const formBlur = ref({
+  email: false,
+  password: false,
+});
+
+const touchForm = () => {
+  touchAll(formBlur.value);
+};
+
+const emailValid = computed(() => {
+  return validateEmail(loginFormData.value.email);
+});
+const passwordValid = computed(() => {
+  return !!loginFormData.value.password;
+});
+
+const emailValidation = computed(() => {
+  return !emailValid.value && formBlur.value.email
+    ? "Please provide a valid email address"
+    : "";
+});
+const passwordValidation = computed(() => {
+  return !passwordValid.value && formBlur.value.password
+    ? "Please provide a valid password"
+    : "";
+});
+
+const submitForm = () => {
+  touchForm();
+  if (!emailValid.value || !passwordValid.value) return;
+  userStore.login({
+    email: loginFormData.value.email,
+    password: loginFormData.value.password,
+  });
+};
+</script>
+
+<template>
+  <div class="login">
+    <h1 class="login__header heading-1">
+      Hi there!<br /><span class="gray">Care to log in?</span>
+    </h1>
+    <BaseCard class="login__form-wrapper">
+      <form
+        action=""
+        class="login__form"
+        novalidate
+        @submit.prevent="submitForm"
+      >
+        <BaseInput
+          input-name="email"
+          v-model="loginFormData.email"
+          placeholder="Something ending with monterail.com"
+          :valid="emailValid"
+          :validation="emailValidation"
+          @blur="formBlur.email = true"
+          >Email</BaseInput
+        >
+        <BaseInput
+          input-name="password"
+          v-model="loginFormData.password"
+          placeholder="Enter your password"
+          input-type="password"
+          :valid="passwordValid"
+          :validation="passwordValidation"
+          @blur="formBlur.password = true"
+          >Password</BaseInput
+        >
+        <div class="login__actions">
+          <BaseButton button-size="large" button-type="secondary"
+            >Register instead</BaseButton
+          >
+          <BaseButton
+            button-size="large"
+            button-type="primary"
+            button-kind="submit"
```

Comment:
Consider replacing it at some point with this:
```suggestion
            size="large"
            type="primary"
            kind="submit"
```

Since it's a `BaseButton` component, naming props beginning with `button-` is redundant, we know it's a button based on component name